### ISSUE TITLE
Piping player output to a stream

### ISSUE THREAD

**Issue by Chris54721** [FrontierPsychiatrist/node-spotify#101]
Hi, is there a way to pipe the output of `spotify.player.play(track)` to a duplex stream?
This is my code ATM:
```
var ss = new stream.PassThrough();
resume = spotify.useNodejsAudio((err, buffer) => {
    if (err) throw err;
    ss.write(buffer);
    return true;
});
```

When I start playback, I get a JS stacktrace:
```
Stacktrace (bbbbbbbb-bbbbbbbb) 0x1a5f58fbef11 0x0: 
==== JS stack trace =========================================

Security context: 0x1a5f58fe3ac1 <JS Object>
    1: DefaultString [native runtime.js:532] [pc=0x3ff8f0db995f] (this=0x1a5f58ffd119 <JS Global Object>,h=0x1a5f58fbef11 <FixedArray[140]>)
    2: ToString(aka ToString) [native runtime.js:~453] [pc=0x3ff8f0e88f36] (this=0x1a5f58f04189 <undefined>,h=0x1a5f58fbef11 <FixedArray[140]>)

==== Details ================================================

[1]: DefaultString [native runtime.js:532] [pc=0x3ff8f0db995f] (this=0x1a5f58ffd119 <JS Global Object>,h=0x1a5f58fbef11 <FixedArray[140]>) {
  // stack-allocated locals
  var L = 0x1a5f58f04189 <undefined>
  var M = 0x1a5f58f04189 <undefined>
  var J = 0x1a5f58f04189 <undefined>
  var K = 0x1a5f58f04189 <undefined>
  // expression stack (top to bottom)
  [07] : 0x1f2f18067e1 <FixedArray[18]>
  [06] : 8
  [05] : 0x1a5f58f56341 <String[8]: toString>
  [04] : 0x1a5f58fbef11 <FixedArray[140]>
--------- s o u r c e   c o d e ---------
function DefaultString(h){\x0aif(!(%_ClassOf(h)==='Symbol')){\x0aif((typeof(h)==='symbol'))throw MakeTypeError(116);\x0avar L=h.toString;\x0aif((%_ClassOf(L)==='Function')){\x0avar M=%_CallFunction(h,L);\x0aif(IsPrimitive(M))return M;\x0a}\x0avar J=h.valueOf;\x0aif((%_ClassOf(J)==='Function')){\x0avar K=%_CallFunction(h,J);\x0aif(IsPrimitive(K))return K...

-----------------------------------------
}

[2]: ToString(aka ToString) [native runtime.js:~453] [pc=0x3ff8f0e88f36] (this=0x1a5f58f04189 <undefined>,h=0x1a5f58fbef11 <FixedArray[140]>) {
// optimized frame
--------- s o u r c e   c o d e ---------
function ToString(h){\x0aif((typeof(h)==='string'))return h;\x0aif((typeof(h)==='number'))return %_NumberToString(h);\x0aif((typeof(h)==='boolean'))return h?'true':'false';\x0aif((h===(void 0)))return'undefined';\x0areturn((h===null))?'null':ToString(DefaultString(h));\x0a}
-----------------------------------------
}
=====================


[1]    26441 illegal hardware instruction  npm start
```

Node.js 5.11.1

**Comment by FrontierPsychiatrist** [comment]
Hm, you should be able to do what you want with the buffer. This should not happen.

**Comment by FrontierPsychiatrist** [comment]
Can reproduce, but no idea what that is yet. Have to investigate.

**Comment by Chris54721** [comment]
Changing the return value to `false` doesn't crash the process, but it still doesn't work as expected. By the way, according to the [docs](https://nodejs.org/api/stream.html#stream_writable_write_chunk_encoding_callback) the `ss.write` method should be async, so the previous code would become:
```
var resume;
var ss = new stream.PassThrough();
resume = spotify.useNodejsAudio((err, buffer) => {
    if (err) throw err;
    ss.write(buffer, 'utf8', resume(true));
    return false;
});
```

**Comment by FrontierPsychiatrist** [comment]
Seems to be the usage of `static` here: https://github.com/FrontierPsychiatrist/node-spotify/blob/1bc3fab7fbc9d3db01a51f048f234ff02eb6edcb/src/audio/NodeAudioHandler.cc#L55

I did that to save the creation of new v8 strings all the time, but I guess that doesn't work anymore. Probably because the handles will be attached to the HandleScope in the first call and then something bad happens in the second call.

Easy fix is to remove `static`, don't know if this will have a bad effect on garbage collection and if how I can mitigate that.

I guess I will start with the easy fix first and release a fix soon.

**Comment by FrontierPsychiatrist** [comment]
Merged fix into development. Will be released with 0.8.
